src/MediawikiApi.php

```diff
@@ -79,15 +79,30 @@ public static function newFromApiEndpoint( $apiEndpoint ) {
 	 * @throws RsdException If the RSD URL could not be found in the page's HTML.
 	 */
 	public static function newFromPage( $url ) {
```

Comment:
RSD could conceivably be done in its own package, then here we'd just have to do something like

```php
$rsd = new Rsd($url);
if (!$rsd->hasApi('MediaWiki')) {
	throw new RsdException( "Unable to find RSD URL in page: $url" );
}
return self::newFromApiEndpoint( $rsd->getApi('MediaWiki')->getApiLink() );
```

But I reckon methods that can be seen whole on one screen are okay. Although I know some people say 20 lines is about the max.

Up to you; I'm happy to rework. (I don't think splitting bits into other methods in this class is much better than this though.)